docs/gallery/howto/autogen/annotate_inputs_outputs.py

```diff
@@ -68,33 +75,50 @@ def add_multiply_graph(x, y):
 wg.run()
 
 # %%
+#
+# .. note::
+#
+#    In ``add_multiply2``, we also annotated the input types (``x: int, y: int``).
+#    This adds a layer of validaton to ensure only integers are passed to the task.
+#    This feature is experimental - its API and behavior may change in future releases.
+#
 # Let's visualize the data provenance of our executed workflow:
 
 generate_node_graph(wg.pk)
 
 # %%
-# As the provenance graph shows, `add_multiply1` has a single output node (`result`), while `add_multiply2` has two separate output nodes (`sum` and `product`), as defined in its namespace.
+# As the provenance graph shows, ``add_multiply1`` has a single output node (``result``), while ``add_multiply2`` has two separate output nodes (``sum`` and ``product``), as defined in its namespace.
 #
 # Static namespaces for inputs
-# -----------------------------
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
 # Similarly, we can annotate inputs to unpack a dictionary into distinct data nodes.
-# We use Python's standard `typing.Annotated` to attach the `aiida-workgraph` namespace metadata to the input type.
-
-from typing import Annotated
+# We use Python's standard ``typing.Annotated`` to attach the `aiida-workgraph` namespace metadata to the input type.
 
 
-@task(outputs=ns(sum=int, product=int))
-def add_multiply3(data: Annotated[dict, ns(x=int, y=int)]):
-    """Takes a dictionary as input but treats 'x' and 'y' as separate nodes."""
+@task
+def add_multiply3(
+    data: t.Annotated[
+        dict,
+        namespace(x=int, y=int),
+    ],
+) -> t.Annotated[
+    dict,
+    namespace(sum=int, product=int),
+]:
+    """Take a dictionary as input but treat 'x' and 'y' as separate nodes."""
     return {"sum": data["x"] + data["y"], "product": data["x"] * data["y"]}
 
 
-@task.graph()
-def add_multiply_graph_inputs(x, y):
-    add_multiply3(data={"x": x, "y": y})
+@task.graph
+def add_multiply_graph_inputs(x: int, y: int):
```

Comment:
Otherwise, there is no link between the task and the graph_outputs.
```suggestion
def add_multiply_graph_inputs(x: int, y: int) -> Annotated[dict, add_multiply3.outptus]:
```

Reply:
Can I really do
`-> Annotated[dict, add_multiply3.outputs]`
or did you mean
`-> Annotated[dict, namespace(add_multiply3=add_multiply3.outputs)]`

Reply:
Also, in

```python
@task
def generate_square_numbers(
    n: int,
) -> t.Annotated[
    dict,
    dynamic(t.Any),
]:
    """Generate a dict of square numbers. The number of outputs depends on 'n'."""
    return {f"square_{i}": i**2 for i in range(n)}


@task.graph
def generate_square_numbers_graph(n: int):
    generate_square_numbers(n=n)


wg = generate_square_numbers_graph.build_graph(n=5)
wg.run()
```

I guess I'm missing a `return` on the graph, but also do I need to again annotate here?

Reply:
Tested, and indeed
`-> Annotated[dict, add_multiply3.outputs]`
works. Very nice!

Reply:
```python
@task.graph
def SquareNumbersGenerator(
    n: int,
) -> t.Annotated[
    dict,
    generate_square_numbers.outputs,
]:
    return generate_square_numbers(n=n)
```

This didn't work, however. The task outputs are not correctly exposed to the graph.